Add new workflow for edgex-cli

tonyespy opened the pull request:
Here are my changes to add the new edgex-cli workflow.

Is there anyway for me to test these changes locally from my repository?

Commit:
Add new workflow for edgex-cli

siggiskulason commented on .github/workflows/wf-edgex-cli.yml:
```diff
@@ -0,0 +1,69 @@
+      - name: Sync and version repo
+        uses: canonical/edgex-sync-and-create-launchpad-branch-action@v1.4.0
+        with:
+          edgex-repo: "edgex-go"
+          canonical-repo: "edgex-go"
```
This was my mistake - I left this from the instructions and will update the CI document. These two lines - "edgex-repo" and "canonical-repo" should be set to the name of the upstream and canonical repositories. in both cases they should be "edgex-cli" instead of "edgex-go"

siggiskulason reviewed (commented):
Other than the comment above, this looks fine to me. There is a way to test things from your own repository. In the past I committed things directly to master while bootstrapping this and setting it up, but now that we have it in place and are using the pull requests and reviews we need to be able to test the workflows. You would need to enable workflows in your own repository, and set up keys in it to be able to access the canonical mirror. I'll document that as well in the CI document.

siggiskulason commented:
Could you also please give the Canonical EdgeX Team admin rights to the edgex-cli repository? That's another step I have to add to the CI document. To do so, go to https://github.com/orgs/canonical/teams/edgex/repositories ,  To the right of the edgex-cli repository select "Admin" instead of read". I see you've already added the repository to the team, but I'll also document the requirement to do that, by entering the repo name in "Find a Repository", selecting it and then clicking "Add Repository".

tonyespy commented:
@siggiskulason thanks for the review. Note, since you actually pointed out at least two changes that need to be addressed, you should've used "Request changes", which would block us from merging this PR until I addressed the issues, and you, or someone else re-approved.

tonyespy commented:
> Could you also please give the Canonical EdgeX Team admin rights to the edgex-cli repository? That's another step I have to add to the CI document. To do so, go to https://github.com/orgs/canonical/teams/edgex/repositories , To the right of the edgex-cli repository select "Admin" instead of read". I see you've already added the repository to the team, but I'll also document the requirement to do that, by entering the repo name in "Find a Repository", selecting it and then clicking "Add Repository".

Done

tonyespy commented:
And requested changes pushed...

siggiskulason commented:
> And requested changes pushed...

Thanks, @tonyespy , LGTM
